### ISSUE TITLE
Problem with inlined (pre-rendered) component and dependencies (v0.110+)

### ISSUE THREAD

**Issue by PavelPancocha** [EmilStenstrom/django-components#789]
Hi!

First thanks for all nice progress in last days!

After a recent update to v0.110+ (v0.113) I found error, which I am not happy with and I am not able to solve it. Maybe there is some magic trick I misseen?

Historically we had some html-fragments rendered directly in python, then moved them to "html-fragments" and rendered still in methods and now we switched that to components. Eg. if you want to show some components in django-admin changelist you do not have much more options.

So we have component, we call it in method with render -> display that in django admin changelist.

Before this change the fragment looked liked this:

![image](https://github.com/user-attachments/assets/3ebe596e-1df3-4775-bbac-5fc2b3482b71)

But now it is like this:

![image](https://github.com/user-attachments/assets/820782fb-937f-4c1f-9c07-9bc9df1960dd)

Is that on purpose? Is this really the best way how to do that? We have two badges, per row, 50 rows = this code is there 100x + the tendencies are injected on the top.

Can I somehow render just the component **without** dependencies?

Thank you!

**Comment by PavelPancocha** [comment]
Hi, thanks for swift reaction.

First things first: `render_dependencies=False` works, now it does the same thing as before. 

I will try to share some code with you, lets start from the bottom:

```python
def get_badge(label: StrOrPromise, color: BadgeColor = BadgeColor.INFO) -> SafeString:
    """
    Return an html badge.
    Color options: primary, success, info, warning, danger
    """
    from components.badge import Badge

    return mark_safe(  # nosec
        Badge.render(
            kwargs={"style": color.value},
            slots={"content": label},
            render_dependencies=False
        )
    )
```

We than use it (with some method in between that just fill the label and color depending on conditions) in admin as method, to be used to display that in admin (in changelist or change page):

```python
    @display(description=_("State"))
    def state(self, obj: CustomUser) -> SafeString:
        """
        Return user’s state as badge.
        """
        return get_user_state_badge(obj)

```

The code is partly historic and being transformed to use components step by step. But the usage pattern seems complete normal for django admin. Feel free to suggest better solution, if you see some! :)

**Comment by JuroOravec** [comment]
@PavelPancocha Glad to hear that you're rendering the badge with `Badge.render()`. If it was embedded inside a template as `{% component %}`, that'd be more difficult to migrate.

Btw where / how do you insert the `badge/style.css` into the HML?

**Comment by JuroOravec** [comment]
On another note, if somebody else has issues with migrating the JS / CSS to v0.110+, please post it here so we can help!

**Comment by PavelPancocha** [comment]
> @PavelPancocha Glad to hear that you're rendering the badge with `Badge.render()`. If it was embedded inside a template as `{% component %}`, that'd be more difficult to migrate.
> 
> Btw where / how do you insert the `badge/style.css` into the HML?

I have it in meta and it is inserted to body. I removed the tag. Seems fine.

**Comment by JuroOravec** [comment]
I just released [v0.117](https://github.com/EmilStenstrom/django-components/releases/tag/0.117), and with that it should be now safe to update from pre-0.110. With that I'll close this issue.

P.S.: @PavelPancocha In [v0.116](https://github.com/EmilStenstrom/django-components/releases/tag/0.116) and later, the extra `<script>` tag that you've mentioned in your first post is now shorter, and it removes itself to clean up the DOM.
